frontend/src/scenes/notebooks/Notebook/Editor.tsx

```diff
@@ -126,14 +130,42 @@ export function Editor({
                         return true
                     }
 
-                    if (event.dataTransfer.files && event.dataTransfer.files[0]) {
+                    if (!moved && event.dataTransfer.files && event.dataTransfer.files[0]) {
                         // if dropping external files
                         const file = event.dataTransfer.files[0] // the dropped file
 
-                        console.warn('TODO: Dropped file!', file)
-                        // TODO: Detect if it is an image and add image upload handler
+                        posthog.capture('notebook file dropped', { file_type: file.type })
+
+                        if (!file.type.startsWith('image/')) {
+                            lemonToast.warning('Only images can be added to Notebooks at this time.')
+                            return true
+                        }
+
+                        const coordinates = view.posAtCoords({
+                            left: event.clientX,
+                            top: event.clientY,
+                        })
+
+                        if (!coordinates) {
+                            // TODO: Seek to end of document instead
+                            return true
+                        }
+
+                        editor
+                            .chain()
+                            .focus()
+                            .setTextSelection(coordinates.pos)
+                            .insertContent({
+                                type: NotebookNodeType.Image,
+                                attrs: {
+                                    file,
```

Comment:
Is this protected from autosaving the file in the contents elsewhere? It looked like that didn't happen based on a quick test but feels like something we shouldn't allow.

Reply:
It _can_ save the contents but they just get saved as an empty object. It felt safe and much more manageable than _not_ including it as part of the state.